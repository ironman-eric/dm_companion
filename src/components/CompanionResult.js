import dndClass from '../data/dndClass';

function getWeightedRandomItem(items) {
  const weights = items.reduce((acc, item, i) => {
    acc.push(item.weight + (acc[i - 1] ?? 0));
    return acc;
  }, []);
  const random = Math.random() * weights.at(-1);  
  return items[weights.findIndex((weight) => weight > random)].label;
}

function CompanionResult({data}) {
  const actions = dndClass[data];  

  return (    
    <p className="text-s italic">Randomized Action: {getWeightedRandomItem(actions)}</p>    
  );
}

export default CompanionResult;
