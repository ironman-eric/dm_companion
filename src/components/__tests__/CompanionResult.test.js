import React from "react";
import renderer from "react-test-renderer";
import CompanionResult from "../CompanionResult";

describe("CompanionResult", () => {
  it("Matches DOM Snapshot", () => {
    const data = 'Barbarian';

  const domTree = renderer.create(<CompanionResult data={data} />).toJSON();
  expect(domTree).toMatchSnapshot();
});
});
