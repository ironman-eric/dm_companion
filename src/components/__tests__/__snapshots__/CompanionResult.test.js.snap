// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanionResult Matches DOM Snapshot 1`] = `
<p
  className="text-s italic"
>
  Randomized Action: 
  Shove
</p>
`;
